[BUG] v1.15.3 regression: IPs referencing non-exsitenct subnets causes KubeOVN controller to panic and CrashLoopBackOff

### Kube-OVN Version

v1.15.3

### Kubernetes Version

v1.33

### Operation-system/Kernel Version

?

### Description

NPE caused by https://github.com/kubeovn/kube-ovn/pull/6275/changes

IP exists but subnet is deleted causes crahloopbackoff:

```
I0212 08:28:13.386144       7 ip.go:241] handle deleting ip myIP.kube-system
W0212 08:28:13.386199       7 ip.go:250] subnet mySubnet not found for deleting ip myIP.kube-system
E0212 08:28:13.386386       7 panic.go:262] "Observed a panic" panic="runtime error: invalid memory address or nil pointer dereference" panicGoValue="\"invalid memory address or nil pointer dereference\"" stacktrace=<
    goroutine 3050 [running]:
    k8s.io/apimachinery/pkg/util/runtime.logPanic({0xb56610ce41e8, 0x4006ee61e0}, {0xb5661071ba20, 0xb56612887320})
        k8s.io/apimachinery@v0.34.3/pkg/util/runtime/runtime.go:132 +0x94
    k8s.io/apimachinery/pkg/util/runtime.handleCrash({0xb56610ce60a8, 0x400086aaf0}, {0xb5661071ba20, 0xb56612887320}, {0x0, 0x0, 0xb5660d684100?})
        k8s.io/apimachinery@v0.34.3/pkg/util/runtime/runtime.go:107 +0xdc
    k8s.io/apimachinery/pkg/util/runtime.HandleCrashWithContext({0xb56610ce60a8, 0x400086aaf0}, {0x0, 0x0, 0x0})
        k8s.io/apimachinery@v0.34.3/pkg/util/runtime/runtime.go:78 +0x5c
    panic({0xb5661071ba20?, 0xb56612887320?})
        runtime/panic.go:783 +0x120
    github.com/kubeovn/kube-ovn/pkg/controller.isOvnSubnet(...)
        github.com/kubeovn/kube-ovn/pkg/controller/subnet.go:1885
    github.com/kubeovn/kube-ovn/pkg/controller.(Controller).handleUpdateIP(0x400042b008, {0x4004c63040?, 0x0?})
        github.com/kubeovn/kube-ovn/pkg/controller/ip.go:261 +0x278
    github.com/kubeovn/kube-ovn/pkg/controller.processNextWorkItem[...].func1(0xb56610c249c0, {0xb5660f6542d4, 0x9?}, 0xb56610c98910, 0xb56610ca92a0, {0x4004c63040, 0x39})
        github.com/kubeovn/kube-ovn/pkg/controller/controller.go:1544 +0x84
    github.com/kubeovn/kube-ovn/pkg/controller.processNextWorkItem...
        github.com/kubeovn/kube-ovn/pkg/controller/controller.go:1550 +0x6c
    github.com/kubeovn/kube-ovn/pkg/controller.(Controller).startWorkers.runWorker[...].func69()
        github.com/kubeovn/kube-ovn/pkg/controller/controller.go:1580 +0x68
    k8s.io/apimachinery/pkg/util/wait.BackoffUntil.func1({0x4013e2a6c8?, 0xb5660d65a874?})
        k8s.io/apimachinery@v0.34.3/pkg/util/wait/backoff.go:233 +0x24
    k8s.io/apimachinery/pkg/util/wait.BackoffUntilWithContext.func1({0xb56610ce60a8?, 0x400086aaf0?}, 0xb56610c249c0?)
        k8s.io/apimachinery@v0.34.3/pkg/util/wait/backoff.go:255 +0x58
    k8s.io/apimachinery/pkg/util/wait.BackoffUntilWithContext({0xb56610ce60a8, 0x400086aaf0}, 0x401607bf38, {0xb56610caab60, 0x4006ee6150}, 0x1)
        k8s.io/apimachinery@v0.34.3/pkg/util/wait/backoff.go:256 +0xac
    k8s.io/apimachinery/pkg/util/wait.BackoffUntil(0xb56610c249c0?, {0xb56610caab60?, 0x4006ee6150?}, 0x8?, 0xb56610c249c0?)
        k8s.io/apimachinery@v0.34.3/pkg/util/wait/backoff.go:233 +0x4c
    k8s.io/apimachinery/pkg/util/wait.JitterUntil(0x400935cec0, 0x3b9aca00, 0x0, 0x1, 0x400086aaf0)
        k8s.io/apimachinery@v0.34.3/pkg/util/wait/backoff.go:210 +0x7c
    k8s.io/apimachinery/pkg/util/wait.Until(...)
        k8s.io/apimachinery@v0.34.3/pkg/util/wait/backoff.go:163
    created by github.com/kubeovn/kube-ovn/pkg/controller.(*Controller).startWorkers in goroutine 165
        github.com/kubeovn/kube-ovn/pkg/controller/controller.go:1429 +0x4d88
 >
panic: runtime error: invalid memory address or nil pointer dereference [recovered, repanicked]
[signal SIGSEGV: segmentation violation code=0x1 addr=0x180 pc=0xb5660f473dc8]
```

### Steps To Reproduce

Create subnet, IP, delete subnet (but force removing finaliser, say)
Restart KubeOVN

### Current Behavior

KubeOVN controller crashes

### Expected Behavior

KubeOVN controller gracefully does not crash, and successfully deletes IP

--- a/pkg/controller/controller_test.go
+++ b/pkg/controller/controller_test.go
@@ -34,6 +34,7 @@ import (
 	kubeovnfake "github.com/kubeovn/kube-ovn/pkg/client/clientset/versioned/fake"
 	kubeovninformerfactory "github.com/kubeovn/kube-ovn/pkg/client/informers/externalversions"
 	kubeovninformer "github.com/kubeovn/kube-ovn/pkg/client/informers/externalversions/kubeovn/v1"
+	ovnipam "github.com/kubeovn/kube-ovn/pkg/ipam"
 	"github.com/kubeovn/kube-ovn/pkg/util"
 )
 
@@ -41,6 +42,7 @@ type fakeControllerInformers struct {
 	vpcInformer       kubeovninformer.VpcInformer
 	vpcNatGwInformer  kubeovninformer.VpcNatGatewayInformer
 	subnetInformer    kubeovninformer.SubnetInformer
+	ipInformer        kubeovninformer.IPInformer
 	serviceInformer   coreinformers.ServiceInformer
 	namespaceInformer coreinformers.NamespaceInformer
 	podInformer       coreinformers.PodInformer
@@ -57,6 +59,7 @@ func alwaysReady() bool { return true }
 // FakeControllerOptions holds optional parameters for creating a fake controller
 type FakeControllerOptions struct {
 	Subnets            []*kubeovnv1.Subnet
+	IPs                []*kubeovnv1.IP
 	NetworkAttachments []*nadv1.NetworkAttachmentDefinition
 	Pods               []*corev1.Pod
 	Namespaces         []*corev1.Namespace
@@ -110,6 +113,13 @@ func newFakeControllerWithOptions(t *testing.T, opts *FakeControllerOptions) (*f
 			return nil, err
 		}
 	}
+	for _, ip := range opts.IPs {
+		_, err := kubeovnClient.KubeovnV1().IPs().Create(
+			context.Background(), ip, metav1.CreateOptions{})
+		if err != nil {
+			return nil, err
+		}
+	}
 
 	// Create informer factories
 	kubeInformerFactory := informers.NewSharedInformerFactoryWithOptions(kubeClient, 0,
@@ -140,12 +150,14 @@ func newFakeControllerWithOptions(t *testing.T, opts *FakeControllerOptions) (*f
 	)
 	vpcInformer := kubeovnInformerFactory.Kubeovn().V1().Vpcs()
 	subnetInformer := kubeovnInformerFactory.Kubeovn().V1().Subnets()
+	ipInformer := kubeovnInformerFactory.Kubeovn().V1().IPs()
 	vpcNatGwInformer := kubeovnInformerFactory.Kubeovn().V1().VpcNatGateways()
 
 	fakeInformers := &fakeControllerInformers{
 		vpcInformer:       vpcInformer,
 		vpcNatGwInformer:  vpcNatGwInformer,
 		subnetInformer:    subnetInformer,
+		ipInformer:        ipInformer,
 		serviceInformer:   serviceInformer,
 		namespaceInformer: namespaceInformer,
 		podInformer:       podInformer,
@@ -156,17 +168,21 @@ func newFakeControllerWithOptions(t *testing.T, opts *FakeControllerOptions) (*f
 
 	// Create controller with all informers
 	ctrl := &Controller{
-		servicesLister:        serviceInformer.Lister(),
-		namespacesLister:      namespaceInformer.Lister(),
-		podsLister:            podInformer.Lister(),
-		vpcsLister:            vpcInformer.Lister(),
-		vpcSynced:             alwaysReady,
-		subnetsLister:         subnetInformer.Lister(),
-		subnetSynced:          alwaysReady,
-		netAttachLister:       nadInformer.Lister(),
-		netAttachSynced:       alwaysReady,
-		OVNNbClient:           mockOvnClient,
-		syncVirtualPortsQueue: newTypedRateLimitingQueue[string]("SyncVirtualPort", nil),
+		servicesLister:          serviceInformer.Lister(),
+		namespacesLister:        namespaceInformer.Lister(),
+		podsLister:              podInformer.Lister(),
+		vpcsLister:              vpcInformer.Lister(),
+		vpcSynced:               alwaysReady,
+		subnetsLister:           subnetInformer.Lister(),
+		subnetSynced:            alwaysReady,
+		ipsLister:               ipInformer.Lister(),
+		ipSynced:                alwaysReady,
+		netAttachLister:         nadInformer.Lister(),
+		netAttachSynced:         alwaysReady,
+		OVNNbClient:             mockOvnClient,
+		ipam:                    ovnipam.NewIPAM(),
+		syncVirtualPortsQueue:   newTypedRateLimitingQueue[string]("SyncVirtualPort", nil),
+		updateSubnetStatusQueue: newTypedRateLimitingQueue[string]("UpdateSubnetStatus", nil),
 	}
 
 	ctrl.config = &Configuration{

--- a/pkg/controller/ip_test.go
+++ b/pkg/controller/ip_test.go
@@ -1,0 +1,50 @@
+package controller
+
+import (
+	"testing"
+
+	"github.com/stretchr/testify/require"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	kubeovnv1 "github.com/kubeovn/kube-ovn/pkg/apis/kubeovn/v1"
+	"github.com/kubeovn/kube-ovn/pkg/util"
+)
+
+func Test_handleUpdateIP_deletedSubnet(t *testing.T) {
+	t.Parallel()
+
+	now := metav1.Now()
+	ip := &kubeovnv1.IP{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:              "test-ip",
+			DeletionTimestamp: &now,
+			Finalizers:        []string{util.KubeOVNControllerFinalizer},
+		},
+		Spec: kubeovnv1.IPSpec{
+			Subnet:    "deleted-subnet",
+			Namespace: "default",
+			PodName:   "test-pod",
+		},
+	}
+
+	fakeCtrl, err := newFakeControllerWithOptions(t, &FakeControllerOptions{
+		IPs: []*kubeovnv1.IP{ip},
+	})
+	require.NoError(t, err)
+
+	ctrl := fakeCtrl.fakeController
+
+	// Shut down work queues to avoid goroutine leaks
+	t.Cleanup(func() {
+		ctrl.updateSubnetStatusQueue.ShutDown()
+		ctrl.syncVirtualPortsQueue.ShutDown()
+	})
+
+	// The subnet "deleted-subnet" does not exist in the fake client.
+	// This must not panic (previously caused NPE in isOvnSubnet).
+	err = ctrl.handleUpdateIP("test-ip")
+	require.NoError(t, err)
+
+	// Verify the subnet status update was enqueued
+	require.Equal(t, 1, ctrl.updateSubnetStatusQueue.Len())
+}

--- a/pkg/controller/subnet.go
+++ b/pkg/controller/subnet.go
@@ -1882,7 +1882,7 @@ func (c *Controller) releaseMcastQuerierIP(subnet *kubeovnv1.Subnet) (bool, erro
 }
 
 func isOvnSubnet(subnet *kubeovnv1.Subnet) bool {
-	return util.IsOvnProvider(subnet.Spec.Provider)
+	return subnet != nil && util.IsOvnProvider(subnet.Spec.Provider)
 }
 
 func formatExcludeIPRanges(subnet *kubeovnv1.Subnet) {

--- a/pkg/controller/subnet_test.go
+++ b/pkg/controller/subnet_test.go
@@ -275,3 +275,46 @@ func Test_formatSubnet(t *testing.T) {
 		})
 	}
 }
+
+func Test_isOvnSubnet(t *testing.T) {
+	t.Parallel()
+
+	tests := []struct {
+		name   string
+		subnet *kubeovnv1.Subnet
+		want   bool
+	}{
+		{
+			name:   "nil subnet returns false",
+			subnet: nil,
+			want:   false,
+		},
+		{
+			name: "empty provider defaults to OVN",
+			subnet: &kubeovnv1.Subnet{
+				Spec: kubeovnv1.SubnetSpec{Provider: ""},
+			},
+			want: true,
+		},
+		{
+			name: "explicit OVN provider",
+			subnet: &kubeovnv1.Subnet{
+				Spec: kubeovnv1.SubnetSpec{Provider: util.OvnProvider},
+			},
+			want: true,
+		},
+		{
+			name: "non-OVN provider",
+			subnet: &kubeovnv1.Subnet{
+				Spec: kubeovnv1.SubnetSpec{Provider: "external.provider"},
+			},
+			want: false,
+		},
+	}
+
+	for _, tc := range tests {
+		t.Run(tc.name, func(t *testing.T) {
+			require.Equal(t, tc.want, isOvnSubnet(tc.subnet))
+		})
+	}
+}

